cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -109,11 +122,20 @@ public ScheduleTaskRunner(Store store, ProgramLifecycleService lifecycleService,
   private ListenableFuture<?> execute(final Id.Program id, Map<String, String> sysArgs,
                                       Map<String, String> userArgs) throws Exception {
     ProgramRuntimeService.RuntimeInfo runtimeInfo;
+    String originalUserId = SecurityRequestContext.getUserId();
     try {
+      // if the program has a namespace user configured then set that user in the security request context.
+      // See: CDAP-7396
+      String nsPrincipal = namespaceQueryAdmin.get(id.getNamespace()).getConfig().getPrincipal();
+      if (nsPrincipal != null && SecurityUtil.isKerberosEnabled(cConf)) {
+        SecurityRequestContext.setUserId(new KerberosName(nsPrincipal).getServiceName());
```

Comment:
generally, the convention we follow is that the state change (in this case setting the user id) which needs to be reset in a finally, happens outside the ``try``, right?. So should we keep this block as:
```
String originalUserId = SecurityRequestContext.getUserId();
try {
  // if the program has a namespace user configured then set that user in the security request context.
  // See: CDAP-7396
  String nsPrincipal = namespaceQueryAdmin.get(id.getNamespace()).getConfig().getPrincipal();
  if (nsPrincipal != null && SecurityUtil.isKerberosEnabled(cConf)) {
    SecurityRequestContext.setUserId(new KerberosName(nsPrincipal).getServiceName());
  }
  try {
    runtimeInfo = lifecycleService.start(id.toEntityId(), sysArgs, userArgs, false);
  } finally {
    if (SecurityUtil.isKerberosEnabled(cConf) {
      SecurityRequestContext.setUserId(originalUserId);
    }
  }
} catch (...) {
...
}
```


Although this is only a nit, feel free to ignore. The only drawback is that the way it is right now, the finally block may get executed more often than necessary